pymongo/change_stream.py

```diff
@@ -422,17 +422,17 @@ def try_next(self) -> Optional[_DocumentType]:
         self._start_at_operation_time = None
 
         if self._decode_custom:
-            return _bson_to_dict(change.raw, self._orig_codec_options)
-        return change
+            return cast(_DocumentType, _bson_to_dict(change.raw, self._orig_codec_options))
+        return cast(_DocumentType, change)
```

Comment:
Can we avoid these casts by defining _orig_codec_options in terms of _DocumentType? casts are bad in my opinion because they can mask legitimate typing bugs so we should only use them as a last resort.

Reply:
Fixed.  However, I noticed that `with_options` should defer to the `CodecOptions` type, since that is often the option you are changing.  This requires casting the parent type.